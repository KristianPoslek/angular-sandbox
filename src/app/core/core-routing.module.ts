import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';

const routes: Routes = [
    {
        path: '',
        loadChildren: '../pokemon-list/pokemon-list.module#PokemonListModule'
    },
    {
        path: 'pokemon',
        loadChildren:
            '../pokemon-details/pokemon-details.module#PokemonDetailsModule'
    }
];

@NgModule({
    imports: [RouterModule.forRoot(routes, { useHash: true })],
    exports: [RouterModule]
})
export class CoreRoutingModule {}
