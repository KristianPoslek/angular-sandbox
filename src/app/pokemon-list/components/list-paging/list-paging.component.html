<nav class="level">
    <div class="level-left">
        <div class="level-item">
            <button
                *ngIf="previousUrl"
                (click)="pagingClicked(this.previousUrl)"
                class="button is-primary"
            >
                Previous
            </button>
        </div>
    </div>

    <div class="level-right">
        <p class="level-item">
            <button
                *ngIf="nextUrl"
                (click)="pagingClicked(this.nextUrl)"
                class="button is-primary"
            >
                Next
            </button>
        </p>
    </div>
</nav>
