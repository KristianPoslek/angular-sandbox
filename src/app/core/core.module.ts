import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule } from '@ngrx/store';
import { PokemonDetailsModule } from '../pokemon-details/pokemon-details.module';
import { PokemonListModule } from '../pokemon-list/pokemon-list.module';
import { SharedModule } from '../shared/shared.module';
import { AppComponent } from './containers/app/app.component';
import { CoreRoutingModule } from './core-routing.module';
import { metaReducers } from './state/meta.reducers';

@NgModule({
    declarations: [AppComponent],
    imports: [
        BrowserModule,
        CoreRoutingModule,
        SharedModule.forRoot(),
        PokemonListModule,
        PokemonDetailsModule,
        StoreModule.forRoot({}, { metaReducers }),
        EffectsModule.forRoot([])
    ]
})
export class CoreModule {}
