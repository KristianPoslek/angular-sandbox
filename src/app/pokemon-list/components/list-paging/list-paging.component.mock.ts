import { Component, EventEmitter, Input, Output } from '@angular/core';

@Component({
    selector: 'pkmn-list-paging',
    template: './list-paging.component.html'
})
export class MockListPagingComponent {
    @Input()
    previousUrl: string;

    @Input()
    nextUrl: string;

    @Output()
    paging: EventEmitter<string> = new EventEmitter();
}
