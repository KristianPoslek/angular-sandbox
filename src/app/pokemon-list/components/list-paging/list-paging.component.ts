import { Component, EventEmitter, Input, Output } from '@angular/core';

@Component({
    selector: 'pkmn-list-paging',
    templateUrl: './list-paging.component.html'
})
export class ListPagingComponent {
    @Input()
    previousUrl: string;

    @Input()
    nextUrl: string;

    @Output()
    paging: EventEmitter<string> = new EventEmitter();

    pagingClicked(url: string): void {
        this.paging.emit(url);
    }
}
