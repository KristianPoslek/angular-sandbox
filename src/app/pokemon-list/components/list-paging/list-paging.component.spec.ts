import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ListPagingComponent } from './list-paging.component';

describe('pokemonList.ListPagingComponent', () => {
    let testee: ListPagingComponent;
    let fixture: ComponentFixture<ListPagingComponent>;

    beforeEach(async () => {
        TestBed.configureTestingModule({
            declarations: [ListPagingComponent]
        });

        fixture = TestBed.createComponent(ListPagingComponent);
        testee = fixture.componentInstance;
    });

    it('should have a snapshot', () => {
        testee.nextUrl = 'foo';
        testee.previousUrl = 'bar';

        expect(fixture).toMatchSnapshot();
    });

    it('should emit paging clicks', () => {
        testee.paging.emit = jest.fn();
        testee.pagingClicked('foo');

        expect(testee.paging.emit).toHaveBeenCalledWith('foo');
    });
});
