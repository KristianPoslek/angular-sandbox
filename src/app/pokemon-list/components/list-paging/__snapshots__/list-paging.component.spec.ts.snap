// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pokemonList.ListPagingComponent should have a snapshot 1`] = `
<pkmn-list-paging
  nextUrl={[Function String]}
  paging={[Function EventEmitter]}
  previousUrl={[Function String]}
>
  <nav
    class="level"
  >
    <div
      class="level-left"
    >
      <div
        class="level-item"
      >
        
        <button
          class="button is-primary"
        >
           Previous 
        </button>
      </div>
    </div>
    <div
      class="level-right"
    >
      <p
        class="level-item"
      >
        
        <button
          class="button is-primary"
        >
           Next 
        </button>
      </p>
    </div>
  </nav>
</pkmn-list-paging>
`;
